concent_api/core/tests/test_unit_admin_tags.py

```diff
@@ -0,0 +1,150 @@
+
+    store_report_computed_task_as_subtask(
+        current_time,
+        task_id,
+        subtask_id,
+        deadline,
+        next_deadline,
+        subtask_state,
+    )
+
+
+class TestAdminTagsQuerySet(ConcentIntegrationTestCase):
+    @staticmethod
+    def test_that_longest_lasting_subtask_timestamp_is_qiven_when_subtasks_with_download_deadline_or_next_deadline_exists_in_database():
+        with freeze_time():
+            current_time = 1836595582
+            store_force_get_task_result(
+                current_time,
+                deadline =current_time + 16,
+                next_deadline=None,
+                task_id='shorter_1',
+                subtask_state=Subtask.SubtaskState.RESULT_UPLOADED,
+            )
+            store_force_get_task_result(
+                current_time,
+                deadline=current_time + 13,
+                next_deadline=current_time + 14,
+                task_id='shorter_2',
+                subtask_state=Subtask.SubtaskState.FORCING_RESULT_TRANSFER,
+            )
+            store_force_get_task_result(
+                current_time,
+                deadline=current_time + 9,
+                next_deadline=current_time + 10,
+                task_id='shorter_3',
+                subtask_state=Subtask.SubtaskState.FORCING_RESULT_TRANSFER,
+            )
+            store_force_get_task_result(
+                current_time,
+                deadline=current_time + 11,
+                next_deadline=current_time + 12,
+                task_id='shorter_4',
+                subtask_state=Subtask.SubtaskState.FORCING_RESULT_TRANSFER,
+            )
+            store_force_get_task_result(
+                current_time,
+                deadline =current_time + 15,
+                next_deadline=None,
+                task_id='z',
+                subtask_state=Subtask.SubtaskState.RESULT_UPLOADED,
+            )
+            store_force_get_task_result(
+                current_time,
+                deadline =current_time + 60,
+                next_deadline=None,
+                task_id='longest',
+                subtask_state=Subtask.SubtaskState.RESULT_UPLOADED,
+            )
+
+            assert_that(get_longest_lasting_subtask_timestamp()).is_equal_to(1836595742)
+
+    @staticmethod
+    def test_that_longest_lasting_subtask_timestamp_method_returns_none_when_empty_database_is_given():
+        assert_that(get_longest_lasting_subtask_timestamp()).is_equal_to(None)
+
+    @staticmethod
+    def test_that_get_time_until_concent_can_be_shut_down_method_returns_zero_when_empty_database_is_given():
+        assert_that(get_time_until_concent_can_be_shut_down()).is_equal_to(datetime.timedelta(0))
+
+    @staticmethod
+    def test_that_get_time_until_concent_can_be_shut_down_method_returns_time_when_not_outdated_tasks_are_in_database():
+        with freeze_time():
+            current_time = get_current_utc_timestamp()
+
+            store_force_get_task_result(
```

Comment:
Only `current_time = get_current_utc_timestamp()` should be in `freeze_time` context, same below. Off course if there is no other reason for it to be there.